cliquet/tests/test_views_hello.py

```diff
@@ -46,3 +46,12 @@ def test_redirect_to_version(self):
                              'http://localhost/v0/mushrooms')
         finally:
             self.app.RequestClass = original_request_class
+
+    def test_do_not_redirect_to_version_if_disabled_in_settings(self):
+        # GET on the hello view.
+        with mock.patch.dict(
+                self.app.app.registry.settings,
+                [('cliquet.version_prefix_redirect_enabled', False)]):
+            response = self.app.get('/')
```

Comment:
I realize that this is using ``self.app`` as defined in ``support.py``. In other words, this request is sent with a version prefix.

The mock patch dict is performed too late, the app was already initialized with redirection enabled, the mocked settings aren't read.

I suggest that you create another app object from scratch without prefixed request class: 

```python
app = self._get_test_app({''cliquet.version_prefix_redirect_enabled': False})
app.RequestClass = TestRequest  # disable redirect
response = app.get('/')
...
...
```